test/publish_test.rb

```diff
@@ -39,4 +39,36 @@ def test_update!
     @pub.update!
     assert_equal('lab-eel', @pub.Label)
   end
+
+  def test_snapshot_kind
+    pub = ::Aptly::PublishedRepository.new(
+      ::Aptly::Connection.new,
+      'Architectures' => ['source'],
+      'Distribution' => 'distro',
+      'Label' => '',
+      'Origin' => '',
+      'Prefix' => 'prefix/kewl-repo-name',
+      'SourceKind' => 'snapshot',
+      'Sources' => [{ 'Component' => 'main', 'Name' => 'kitten' }],
+      'Storage' => ''
+    )
+
+    assert(pub.Sources[0].is_a?(Aptly::Snapshot))
```

Comment:
AFAICT from https://www.aptly.info/doc/api/publish/ , SourceKind is always only a string.